### ISSUE TITLE
 Bad Request: message can't be edited

### ISSUE THREAD

**Issue by fei-ke** [blueset/ehForwarderBot#33]
When occurred, Cannot receive message.
```
plugins.eh_telegram_master.TelegramChannel [ERROR]
    BadRequest()Traceback (most recent call last):
  File "/opt/ehForwarderBot/plugins/eh_telegram_master/__init__.py", line 309, in process_msg
    parse_mode=parse_mode)
  File "/usr/lib/python3.6/site-packages/telegram/bot.py", line 125, in decorator
    result = func(self, *args, **kwargs)
  File "/usr/lib/python3.6/site-packages/telegram/bot.py", line 151, in decorator
    result = self._request.post(url, data, timeout=kwargs.get('timeout'))
  File "/usr/lib/python3.6/site-packages/telegram/utils/request.py", line 207, in post
    **urlopen_kwargs)
  File "/usr/lib/python3.6/site-packages/telegram/utils/request.py", line 152, in _request_wrapper
    raise BadRequest(repr(message))
telegram.error.BadRequest: "Bad Request: message can't be edited"

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/ehForwarderBot/plugins/eh_telegram_master/__init__.py", line 313, in process_msg
    text=msg_template + msg.text)
  File "/usr/lib/python3.6/site-packages/telegram/bot.py", line 125, in decorator
    result = func(self, *args, **kwargs)
  File "/usr/lib/python3.6/site-packages/telegram/bot.py", line 151, in decorator
    result = self._request.post(url, data, timeout=kwargs.get('timeout'))
  File "/usr/lib/python3.6/site-packages/telegram/utils/request.py", line 207, in post
    **urlopen_kwargs)
  File "/usr/lib/python3.6/site-packages/telegram/utils/request.py", line 152, in _request_wrapper
    raise BadRequest(repr(message))
telegram.error.BadRequest: "Bad Request: message can't be edited"
```

**Comment by blueset** [comment]
Please try to describe what triggered this issue in detail.

**Comment by fei-ke** [comment]
@blueset  Probably occurred when I received a WeChat message, and this message doesn't send to my Telegram client.

**Comment by blueset** [comment]
@fei-ke How often do you receive this error? Can you reproduce this error stably?

**Comment by fei-ke** [comment]
@blueset Yesterday got 10 times or so, can't reproduce, just some normal text messages, I will continue to try.

**Comment by martixingwei** [comment]
Yeah, I got this problem too.

**Comment by PeterCxy** [comment]
Can reproduce here too. Seems to be randomly happening.

**Comment by vjdsnagiurt** [comment]
遇到同样的问题, 我可以重现此问题. 
我关注了一个叫"谁是卧底助手"的服务号, 从Bot里用/chat 会话头的方式和它对话, 发"进", 它会回复"请输入房间号:" 这是正常的.
但是, 如果把"谁是卧底助手"/link到一个群组, 在群组里发"进", 就会出现, Telegram的状态栏会显示该Bot正在输入, 但并没有发来任何消息. 去后台查询EFB.log, 显示
telegram.error.BadRequest: "Bad Request: message can't be edited".

我换了两台VPS, 都能重现此问题. 

我使用python3 main.py -vv打开debug模式, 得到如下的信息:

2017-03-27 11:17:33,207: plugins.eh_wechat_slave.WeChatChannel [INFO]
    Sending message to WeChat:
Target-------
uid: 2582032370
UserName: @92b747c26a33166cf1b985651cb5648a
NickName:
Type: Text
Text: 进
2017-03-27 11:17:33,207: itchat [DEBUG]
    Request to send a text message to @92b747c26a33166cf1b985651cb5648a: 进
2017-03-27 11:17:33,510: peewee [DEBUG]
    ('INSERT INTO "msglog" ("master_msg_id", "slave_message_id", "text", "slave_origin_uid", "slave_origin_display_name", "slave_member_uid", "slave_member_display_name", "msg_type", "sent_to", "time") VALUES (?
, ?, ?, ?, ?, ?, ?, ?, ?, ?)', ['-183300378.266', '4434172563418044082', '进', 'eh_wechat_slave.2582032370', '__chat__', None, None, 'Text', 'slave', datetime.datetime(2017, 3, 27, 11, 17, 33, 509250)])
2017-03-27 11:17:34,610: plugins.eh_wechat_slave.wechat_msg_meta [DEBUG]
    Raw message: {'ImgStatus': 1, 'ForwardFlag': 0, 'NewMsgId': 3809041608357499111, 'SubMsgType': 0, 'ImgHeight': 0, 'VoiceLength': 0, 'Content': '请输入房间号:', 'RecommendInfo': {'AttrStatus': 0, 'Signature':
 '', 'Content': '', 'OpCode': 0, 'VerifyFlag': 0, 'Sex': 0, 'City': '', 'Ticket': '', 'UserName': '', 'NickName': '', 'QQNum': 0, 'Scene': 0, 'Alias': '', 'Province': ''}, 'PlayLength': 0, 'ToUserName': '@4c6bad
aa76c57849c0bab618510dc57e', 'FromUserName': '@92b747c26a33166cf1b985651cb5648a', 'Type': 'Text', 'Url': '', 'Status': 3, 'StatusNotifyCode': 0, 'CreateTime': 1490584653, 'FileName': '', 'AppMsgType': 0, 'MsgTyp
e': 1, 'StatusNotifyUserName': '', 'Ticket': '', 'OriContent': '', 'ImgWidth': 0, 'Text': '请输入房间号:', 'AppInfo': {'Type': 0, 'AppID': ''}, 'MediaId': '', 'FileSize': '', 'MsgId': '3809041608357499111', 'Has
ProductId': 0}
2017-03-27 11:17:34,610: plugins.eh_wechat_slave.wechat_msg_meta [DEBUG]
    me, False
2017-03-27 11:17:34,815: plugins.eh_wechat_slave.wechat_msg_meta [DEBUG]
    From user, {'Sex': 0, 'ChatRoomId': 0, 'RemarkPYQuanPin': '', 'SnsFlag': 0, 'UserName': '@92b747c26a33166cf1b985651cb5648a', 'VerifyFlag': 24, 'MemberCount': 0, 'City': '', 'KeyWord': 'gh_', 'Uin': 'gh_2dbfe
254d26d', 'AppAccountFlag': 0, 'OwnerUin': 0, 'MemberList': [], 'ContactFlag': 1, 'Signature': '最省流量的卧底助手，为你出题，帮你发题。 如果线下人不够，我还能帮你在微信里凑局呢~', 'RemarkName': '', 'PYQuanPin':
 'sheishiwodezhushou', 'Statues': 0, 'IsOwner': 0, 'DisplayName': '', 'UniFriend': 0, 'Province': '北京', 'AttrStatus': 0, 'NickName': '谁是卧底助手', 'HeadImgUrl': '/cgi-bin/mmwebwx-bin/webwxgeticon?seq=6511881
33&username=@92b747c26a33166cf1b985651cb5648a&skey=@crypt_343f7d59_cf8485f73ec9c9f08b9574ef978d8b59', 'RemarkPYInitial': '', 'PYInitial': 'SSWDZS', 'EncryChatRoomId': '', 'Alias': 'wodizhushou', 'HideInputBarFla
g': 0, 'StarFriend': 0}
2017-03-27 11:17:35,110: plugins.eh_wechat_slave.wechat_msg_meta [DEBUG]
    dest: {'alias': '匿名', 'name': '匿名', 'uid': '12345678'}
2017-03-27 11:17:35,111: plugins.eh_wechat_slave.wechat_msg_meta [INFO]
    WeChat incoming message:
Type: Text
Text: 请输入房间号:
UserName: @92b747c26a33166cf1b985651cb5648a
uid: 2582032370
name: 谁是卧底助手
2017-03-27 11:17:35,111: plugins.eh_telegram_master.TelegramChannel [INFO]
    Got message from queue
Type: Text
Text: 请输入房间号:
----
2017-03-27 11:17:35,203: plugins.eh_telegram_master.TelegramChannel [DEBUG]
    1490584655.203302, Msg text: 请输入房间号:
2017-03-27 11:17:35,204: plugins.eh_telegram_master.TelegramChannel [INFO]
    Msg sent to TG, task_done marked.
2017-03-27 11:17:35,204: plugins.eh_telegram_master.TelegramChannel [DEBUG]                                                                                                                                 [0/279]
    1490584655.203302, process_msg_step_0
2017-03-27 11:17:35,205: peewee [DEBUG]
    ('SELECT "t1"."id", "t1"."master_uid", "t1"."slave_uid" FROM "chatassoc" AS t1 WHERE ("t1"."slave_uid" = ?)', ['eh_wechat_slave.2582032370'])
2017-03-27 11:17:35,208: peewee [DEBUG]
    ('SELECT "t1"."id", "t1"."master_uid", "t1"."slave_uid" FROM "chatassoc" AS t1 WHERE ("t1"."master_uid" = ?)', ['eh_telegram_master.-183300378'])
2017-03-27 11:17:35,209: plugins.eh_telegram_master.TelegramChannel [DEBUG]
    1490584655.203302, process_msg_step_1, tg_dest=111039943, msg.origin={'alias': '谁是卧底助手', 'name': '谁是卧底助手', 'uid': '2582032370'}
2017-03-27 11:17:35,209: plugins.eh_telegram_master.TelegramChannel [DEBUG]
    1490584655.203302, process_msg_step_2
2017-03-27 11:17:36,504: peewee [DEBUG]
    ('SELECT "t1"."master_msg_id", "t1"."slave_message_id", "t1"."text", "t1"."slave_origin_uid", "t1"."slave_origin_display_name", "t1"."slave_member_uid", "t1"."slave_member_display_name", "t1"."msg_type", "t1
"."sent_to", "t1"."time" FROM "msglog" AS t1 WHERE ("t1"."master_msg_id" LIKE ?) ORDER BY "t1"."time" DESC LIMIT 1', ['-183300378.%'])
2017-03-27 11:17:36,509: plugins.eh_telegram_master.TelegramChannel [DEBUG]
    Text: Append last msg: True
2017-03-27 11:17:36,510: plugins.eh_telegram_master.TelegramChannel [DEBUG]
    1490584655.203302, process_msg_step_3_0, tg_dest = -183300378, tg_chat_assoced = True, append_last_msg = True
2017-03-27 11:17:36,510: plugins.eh_telegram_master.TelegramChannel [DEBUG]
    1490584655.203302, process_msg_step_3_0_1
2017-03-27 11:17:36,905: plugins.eh_telegram_master.TelegramChannel [ERROR]
    BadRequest()Traceback (most recent call last):
  File "/home/dou/ehForwarderBot/plugins/eh_telegram_master/__init__.py", line 309, in process_msg
    parse_mode=parse_mode)
  File "/home/dou/.pyenv/versions/miniconda2-latest/envs/wechat/lib/python3.5/site-packages/telegram/bot.py", line 125, in decorator
    result = func(self, *args, **kwargs)
  File "/home/dou/.pyenv/versions/miniconda2-latest/envs/wechat/lib/python3.5/site-packages/telegram/bot.py", line 151, in decorator
    result = self._request.post(url, data, timeout=kwargs.get('timeout'))
  File "/home/dou/.pyenv/versions/miniconda2-latest/envs/wechat/lib/python3.5/site-packages/telegram/utils/request.py", line 207, in post
    **urlopen_kwargs)
  File "/home/dou/.pyenv/versions/miniconda2-latest/envs/wechat/lib/python3.5/site-packages/telegram/utils/request.py", line 152, in _request_wrapper
    raise BadRequest(repr(message))
telegram.error.BadRequest: "Bad Request: message can't be edited"

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/dou/ehForwarderBot/plugins/eh_telegram_master/__init__.py", line 313, in process_msg
    text=msg_template + msg.text)
  File "/home/dou/.pyenv/versions/miniconda2-latest/envs/wechat/lib/python3.5/site-packages/telegram/bot.py", line 125, in decorator
    result = func(self, *args, **kwargs)
  File "/home/dou/.pyenv/versions/miniconda2-latest/envs/wechat/lib/python3.5/site-packages/telegram/bot.py", line 151, in decorator
    result = self._request.post(url, data, timeout=kwargs.get('timeout'))
  File "/home/dou/.pyenv/versions/miniconda2-latest/envs/wechat/lib/python3.5/site-packages/telegram/utils/request.py", line 207, in post
    **urlopen_kwargs)
  File "/home/dou/.pyenv/versions/miniconda2-latest/envs/wechat/lib/python3.5/site-packages/telegram/utils/request.py", line 152, in _request_wrapper
    raise BadRequest(repr(message))
telegram.error.BadRequest: "Bad Request: message can't be edited"

**Comment by blueset** [comment]
@vjdsnagiurt Issue reproduced and investigated, thanks for the detailed report.

Released hotfix v1.6.1 (ade7919) attempt to address this issue. Please upgrade to the latest EFB version and restart your EFB instance. If the issue persists, please report according the the contributing guideline in this issue thread.

感谢提供详细的 Bug 报告，问题已在本地重现并尝试修复。

发布了 Hotfix 版本 v1.6.1 (ade7919) 尝试解决该问题。请更新 EFB 至最新版并重启 EFB 以应用更新。如该问题依旧存在，请参照贡献指南并将其报告与此。
